spec/services/v1/question_group/share_externally_service_spec.rb

```diff
@@ -42,4 +42,28 @@
       expect { subject.call }.to raise_error(CanCan::AccessDenied)
     end
   end
+
+  context 'when the invited researcher does not have an activated account' do
+    subject { described_class.new(researcher_emails, source_session_id, selected_groups_with_questions, current_user) }
+
+    let(:new_researcher) { create(:user, :researcher, :unconfirmed) }
+    let!(:researcher_emails) { [new_researcher.email] }
+
+    let(:message_delivery) { instance_double(ActionMailer::MessageDelivery) }
+
+    before do
+      allow(message_delivery).to receive(:deliver_now)
+      ActiveJob::Base.queue_adapter = :test
+    end
+
+    it 'creates a new account for the researcher' do
+      subject.call
+      expect(User.exists?(id: new_researcher.id)).to eq(true)
+    end
```

Comment:
I don't think this test does something. You have in lazy let but when you call it here you create a user at this moment. To sum up -> you don't test your code here.